File: postrgres-ubuntu/db_sample.go
package main

import (
	"database/sql"
	"fmt"
	_ "github.com/lib/pq"
	"os"
	// "time"
)

var (
	DB_USER     = "docker"
	DB_PASSWORD = "docker"
	DB_NAME     = "docker"
	DB_HOST     = "localhost"
	DB_PORT     = os.Getenv("PG_PORT")
)

func main() {
	// dbinfo := fmt.Sprintf("user=%s password=%s host=localhost port=%s dbname=%s sslmode=disable",
	// DB_USER, DB_PASSWORD, DB_NAME, DB_PORT)

	dbinfo := fmt.Sprintf("postgres://%s:%s@%s:%s/%s?sslmode=disable", DB_USER, DB_PASSWORD, DB_HOST, DB_PORT, DB_NAME)
	fmt.Printf("Connecting to %s\n", dbinfo)
	db, err := sql.Open("postgres", dbinfo)
	checkErr(err)
	defer db.Close()

	rows, err := db.Query("SELECT id, name,address FROM COMPANY")
	checkErr(err)
	defer rows.Close()
	fmt.Println("uid | username | address ")
	for rows.Next() {
		var id int
		var name string
		var address string

		err = rows.Scan(&id, &name, &address)
		checkErr(err)
		fmt.Printf("%3v | %8v | %6v \n", id, name, address)
	}
	err = rows.Err()
	checkErr(err)

}

func checkErr(err error) {
	if err != nil {
		panic(err)
	}
}

File: postrgres-ubuntu/db_sample.py
#!/usr/bin/python3

import psycopg2
import os


if __name__ == '__main__':
	
	PG_PORT = os.environ.get("PG_PORT")
	conn = None
	
	try:
     
		conn = psycopg2.connect(database="docker", user="docker", password="docker", host="127.0.0.1", port=PG_PORT)
		print( "Opened database successfully")

		cur = conn.cursor()

		cur.execute("SELECT * FROM company;")
		
		print(" ".join(["{:>12s}".format(i.name) for i in cur.description]))
		rows = cur.fetchall()

		for row in rows:
			print (row)

	except psycopg2.DatabaseError as e:
	    print ('Error {ERR}'.format(ERR=e))
	    sys.exit(1)
    
    
	finally:
	    
	    if conn:
	        conn.close()
